gradle-plugin/src/main/kotlin/com/google/devtools/ksp/gradle/KspSubplugin.kt

```diff
@@ -136,6 +137,9 @@ class KspGradleSubplugin @Inject internal constructor(private val registry: Tool
             kspExtension.apOptions.forEach {
                 options += SubpluginOption("apoption", "${it.key}=${it.value}")
             }
+            kspExtension.commandLineArgumentProviders.forEach {
+                options += SubpluginOption("apoption", it.asArguments().joinToString(""))
```

Comment:
The compiler plugin side assumes each apoption to be of the form "k=v". You probably need one more loop instead of `joinToString("")`. E.g.,
```
kspExtension.commandLineArgumentProviders.forEach {
  it.asArguments.forEach {
    options += SubpluginOption("apoption", it)
  } 
}
```

On the other hand, it looks like that there's no guaranteed form of strings in `CommandlineArgumentProvider.asArguments()`. Can you insert a check here to make sure that all arguments follow the form "k=v"?

Reply:
Sounds good. The minimum check would be that `asArguments().joinToString("")` matches the regex `\S=\S`?

I will add that regex check.

Reply:
regex and test added